File: java/smartlist/src/main/java/com/smartlist/module/grading/dto/request/QuestionCommentReqDTO.java
package com.smartlist.module.grading.dto.request;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotEmpty;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class QuestionCommentReqDTO {

    @Schema(description = "Username", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotEmpty
    String username;

    @Schema(description = "Comment", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotEmpty
    String comment;


}

File: javascript/smartlist/smartlist-api-mysql-project/src/modules/grading/dto/request/question-comment.req.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { IsNotEmpty } from 'class-validator';
export class QuestionCommentReqDto {
  @ApiProperty({
    type: String,
  })
  @IsNotEmpty()
  username: string;

  @ApiProperty({
    type: String,
  })
  @IsNotEmpty()
  comment: string;
}
